java/core/src/java/org/apache/orc/TypeDescription.java

```diff
@@ -420,15 +418,9 @@ public boolean equals(Object other, boolean checkAttributes) {
     }
     if (checkAttributes) {
       // make sure the attributes are the same
-      List<String> attributeNames = getAttributeNames();
```

Comment:
Calling `getAttributeNames()` method here actually causes an ArrayList instantiation, and a sort of that list, for ever invocation.